pkg/driver/node_server.go

```diff
@@ -1801,63 +1801,25 @@ func (driver *Driver) NodeGetVolumeStats(ctx context.Context, in *csi.NodeGetVol
 			in.VolumePath)
 		return nil, nil
 	}
-	pvName := folders[len(folders)-2]
-	credentials, err := driver.flavor.GetCredentialsFromVolume(pvName)
-	if err != nil {
-		log.Error("err: ", err.Error())
-		return nil, status.Errorf(codes.Unavailable, "Failed to get credentials from volume with name %s, err: %s",
-			pvName, err.Error())
-	}
 
-	// Get storage provider using secrets
-	storageProvider, err := driver.GetStorageProvider(credentials)
-	if err != nil {
-		log.Error("err: ", err.Error())
-		return nil, status.Errorf(codes.Unavailable, "Failed to get storage provider from secrets for volume with id %s, err: %s",
-			volumeID, err.Error())
-	}
-	// Fetch the volume using volume ID
-	volume, err := storageProvider.GetVolume(volumeID)
-	if err != nil {
-		log.Error("err: ", err.Error())
-		return nil, status.Errorf(codes.Internal, "Error while retrieving volume with id %s from the backend, err: %s",
-			volumeID, err.Error())
-	}
-	if volume == nil {
-		log.Errorf("Volume with ID %s not found on the backend, return ", volumeID)
-		return nil, status.Errorf(codes.NotFound, "Volume with ID %s not found on the backend, return ",
-			volumeID)
-	}
-	hpeMetricsProvider := vol.NewMetricsStatFS(in.VolumePath)
-	metrics, err := hpeMetricsProvider.GetMetrics()
+	var statfs syscall.Statfs_t
```

Comment:
Ideally, this functionality should be implemented in the `common-host-libs/linux package` and call that method here. In the future, let's plan to move it there.